projects/plugins/social/tests/e2e/README.md

```diff
@@ -0,0 +1,143 @@
+- [Test reports](#test-reports)
+
+## Pre-requisites
+
+Install dependencies and build Jetpack Social first.
+
+```shell
+# run in the monorepo root
+pnpm install
+pnpm jetpack build plugins/social
+```
+
+## Environment setup
+
+### Test configuration
+
+E2E tests rely on an encrypted configuration file, which is included in the [e2e commons package](../../../../../tools/e2e-commons) config folder as [`encrypted.enc`](../../../../../tools/e2e-commons/config/encrypted.enc). To be able to run tests - that file should be decrypted first.
+
+To decrypt the config files (a8c only):
+
+- Find a decryption key. Search secret store for "E2E Jetpack CONFIG_KEY"
+- Run `CONFIG_KEY=YOUR_KEY pnpm config:decrypt`. This command should create a new file [`config/local.cjs`](./config/local.cjs)
+
+### Docker environment
+
+Tests expect to have a WordPress installation with Jetpack installed, accessible via a local tunnel.
+
+To start the environment:
+
+```shell
+pnpm env:up
+```
+
+This will create the Docker environment and configure the WordPress installation. It will start one WordPress container and one database container. The WordPress installation is available at `localhost:8889`.
+
+#### Local tunnel
+
+To bypass the offline mode you will need your site to have a publicly accessible url that will proxy all requests to your locally running WordPress installation.
+These tests use `localtunnel` library to expose `localhost:8889` via a public url.
+
+To start a tunnel
+
+```
+pnpm tunnel:up
+```
+
+To stop the tunnel
+
+```
+pnpm tunnel:down
+```
+
+The tunnel url will be stored in a file so that it can be read by the tests and then reused by the tunnel script. See config files for details.
+
+If you want a different url, use the `reset` command.
+
+```
+pnpm tunnel:reset
+```
+
+## Running tests
+
+Once your target WP environment is running on `localhost:8889` you can run the tests.
+
+Run all tests: `pnpm test:run`
+
+Playwright runs headless by default (i.e. browser is not visible). However, sometimes it's useful to observe the browser while running tests. To see the browser window, and the running tests you can use the `--headed` flag:
+
+```bash
+pnpm test:run --headed
+```
+
+To run an individual test, use the direct path to the spec. For example:
+
+```bash
+pnpm test:run ./specs/connection.test.js
+```
+
+To run in debug mode, use the `--debug` flag. Debug mode uses a headed browser and opens the [Playwright inspector](https://playwright.dev/docs/inspector/).
+
+```bash
+pnpm test:run --debug
+```
+
+### Selecting tests to run
+
+```bash
+# One test file
+pnpm test:run ./specs/connection.test.js
+
+# All tests having 'connection' in their name
+pnpm test:run connection
+
+# Run only run tests matching a regular expression.
+pnpm test:run --grep "connection"
+pnpm test:run -g "connection"
+
+# Run only run tests NOT matching a regular expression.
+pnpm test:run --grep-invert "connection"
+```
+
+## Tests Architecture
```

Comment:
I wonder if we need to duplicate the contents of e2e-commons here. Maybe instead, we can link it to the e2e-commons package?

Reply:
Yep! Link to a single source is much easier to maintain.

Reply:
I don't see it as a duplicate of `e2e-commons`. Part of this is similar with the quick start part in commons, it starts from there, but each project can evolve differently. They can update the package.json scripts or add new ones, have different structure or testing approaches or different CI configs. Some parts may remain similar, but I think this is a case where I prefer some duplication.